core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/HttpCallbackAction.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.action;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.batch.BatchActionContext;
+import co.cask.cdap.etl.api.batch.PostAction;
+import com.google.common.base.Charsets;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.jboss.netty.handler.codec.http.HttpMethod;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.OutputStream;
+import java.lang.reflect.Type;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Makes an HTTP call at the end of a pipeline run.
+ */
+@Plugin(type = PostAction.PLUGIN_TYPE)
+@Name("HttpCallback")
+@Description("Makes an HTTP call at the end of a pipeline run.")
+public class HttpCallbackAction extends PostAction {
+  private static final Logger LOG = LoggerFactory.getLogger(HttpCallbackAction.class);
+  private final Conf conf;
+
+  public HttpCallbackAction(Conf conf) {
+    this.conf = conf;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    conf.validate();
+  }
+
+  @SuppressWarnings("ConstantConditions")
+  @Override
+  public void run(BatchActionContext batchActionContext) throws Exception {
+    int retries = 0;
+    Exception exception = null;
+    do {
+      HttpURLConnection conn = null;
```

Comment:
I don't think its a good idea to pull in cdap-common just for this.  It would add unneeded dependencies, make the jar bigger, and set a bad example for people who are looking at hydrator-plugins by making it look like you need more than just the cdap api to write a plugin